spring-integration-core/src/main/java/org/springframework/integration/aggregator/CorrelatingMessageBarrier.java

```diff
@@ -118,7 +120,7 @@ private Object getLock(Object correlationKey) {
 						Iterator<Message<?>> messages = group.getMessages().iterator();
 						if (messages.hasNext()) {
 							nextMessage = messages.next();
-							store.removeMessageFromGroup(key, nextMessage);
+							this.store.removeMessagesFromGroup(key, Collections.<Message<?>>singletonList(nextMessage));
```

Comment:
Good point, I started with vararg but realized there are some places where we already have a collection; good idea to add both variants.